caddy/bolt_transport.go

```diff
@@ -0,0 +1,128 @@
+package caddy
+
+import (
+	"bytes"
+	"encoding/gob"
+	"strconv"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/dunglas/mercure"
+)
+
+func init() { //nolint:gochecknoinits
+	caddy.RegisterModule(Bolt{})
+}
+
+type Bolt struct {
+	Path             string  `json:"path,omitempty"`
+	BucketName       string  `json:"bucket_name,omitempty"`
+	Size             uint64  `json:"size,omitempty"`
+	CleanupFrequency float64 `json:"cleanup_frequency,omitempty"`
+
+	transport    *mercure.BoltTransport
+	transportKey string
+}
+
+// CaddyModule returns the Caddy module information.
+func (Bolt) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "http.handlers.mercure.bolt",
```

Comment:
Hopefully it's not too late, I would recommend grouping these modules under a namespace like `http.handlers.mercure.transports.*` because as-is if you ever want to modularize something else within mercure then the entire namespace under it is already taken by the transports. So if you make another namespace like `http.handlers.mercure.whatever.*` technically that's inside of `http.handlers.mercure.*` which you're already using now for transports.